mocpy/moc.py

```diff
@@ -2,427 +2,360 @@
-            uniq_interval_add = np.vectorize(uniq_interval.add)
-            uniq_interval_add(utils.orderipix2uniq(n_order, np.array(n_pix_l)))
+        Parameters
+        ----------
+        url : str
+            the url where the fits file representing a moc is available
 
-        return MOC.from_uniq_interval_set(uniq_interval)
+        Returns
+        -------
+        result : `~mocpy.moc.MOC`
+            the created moc
 
-    @classmethod
-    def from_uniq_interval_set(cls, uniq_is):
-        """
-        Create a MOC from an IntervalSet of NUNIQ HEALPix pixels
-        """
-        r = IntervalSet()
-        rtmp = IntervalSet()
-        last_order = 0
-        
-        intervals = uniq_is.intervals
-        max_res_order = MOC.HPY_MAX_NORDER
-        diff_order = max_res_order
-        for interval in intervals:
-            for j in xrange(interval[0], interval[1]):
-                order, ipix = utils.uniq2orderipix(j)
-                
-                if order != last_order:
-                    r = r.union(rtmp)
-                    rtmp.clear()
-                    last_order = order
-                    diff_order = max_res_order - order
-
-                rtmp.add( ( ipix * 4**(diff_order), (ipix+1) * 4**(diff_order) ) )
-
-            first = False
-
-        r = r.union(rtmp)
-        return cls.from_interval_set(r)
-    
-    @classmethod
-    def from_interval_set(cls, interval_set):
-        """
-        Create a MOC from an IntervalSet (all intervals at deepest norder)
         """
-        moc = MOC()
-        moc._interval_set = interval_set
-        
-        return moc
-    
+
+        from astropy.utils.data import download_file
+
+        path = download_file(url, show_progress=False, timeout=60)
+        result = cls.from_moc_fits_file(path)
+        return result
+
     @classmethod
     def from_table(cls, table, ra_column, dec_column, moc_order):
         """
-        Create a MOC from a astropy.table.Table
-        The user has to specify the columns holding ra and dec (in ICRS)
+        Create a `~mocpy.moc.MOC` object from a `~astropy.table.Table`
+
+        The user has to specify the columns holding the ra and dec (in ICRS) data in ``table``
+
+        Parameters
+        ----------
+        table : `~astropy.table.Table`
+            the observations astropy table
+        ra_column : str
+            the name of the column referring to the right ascension of the observations
+        dec_column : str
+            the name of the column referring to the declination of the observations
+        moc_order : int
+            the max order of the MOC that will be created
+
+        Returns
+        -------
+        moc : `~mocpy.moc.MOC`
+            the created moc
+
         """
+
         if not isinstance(ra_column, str) or not isinstance(dec_column, str):
-            raise TypeError
+            raise TypeError('`ra_column` whose type is {0} or `dec_column` whose type is {1} must be of string type'
```

Comment:
Too few arguments for format string. This will error. Suggest to just remove the whole isinstance check.